### PR TITLE
Add bpk-component-definitions-list

### PR DESCRIPTION
**Pull request by janeklb**
Entirely based on `bpk-component-list`, just a simple wrapper around `dl`, `dt`, and `dd`.

### CODE REVIEW

File: packages/bpk-component-definitions-list/src/BpkDefinition.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import STYLES from './bpk-definitions-list.scss';
+
+const getClassName = cssModules(STYLES);
+
+const BpkDefinition = (props) => {
+  const classNames = [getClassName('bpk-definitions-list__definition')];
+
+  if (props.className) { classNames.push(props.className); }
+
+  return (<dd className={classNames.join(' ')}>{props.children}</dd>);
```
Comment (k0nserv) [inline packages/bpk-component-definitions-list/src/BpkDefinition.js]: The parenthesis are redundant here right? Can we remove them?

File: packages/bpk-component-definitions-list/src/BpkDefinition.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+BpkDefinition.propTypes = {
+  children: PropTypes.oneOfType([
```
Comment (k0nserv) [inline packages/bpk-component-definitions-list/src/BpkDefinition.js]:
```javascript
children: PropTypes.node
```

Is enough AFAIK. `PropTypes.node` already covers one vs many

File: packages/bpk-component-definitions-list/src/BpkDefinitionsList.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const BpkDefinitionsList = (props) => {
+  const classNames = [getClassName('bpk-definitions-list')];
+
+  if (props.className) { classNames.push(props.className); }
+
+  return <dl className={classNames.join(' ')}>{props.children}</dl>;
+};
+
+BpkDefinitionsList.propTypes = {
+  children: PropTypes.oneOfType([
```
Comment (k0nserv) [inline packages/bpk-component-definitions-list/src/BpkDefinitionsList.js]: Same as above

File: packages/bpk-component-definitions-list/src/BpkTerm.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const BpkTerm = (props) => {
+  const classNames = [getClassName('bpk-definitions-list__term')];
+
+  if (props.className) { classNames.push(props.className); }
+
+  return (<dt className={classNames.join(' ')}>{props.children}</dt>);
+};
+
+BpkTerm.propTypes = {
+  children: PropTypes.oneOfType([
```
Comment (k0nserv) [inline packages/bpk-component-definitions-list/src/BpkTerm.js]: Same as above

**Comment by k0nserv** [comment]
A docs page hasn't been created. See the [TicketsPage](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-docs/src/pages/TicketsPage/TicketsPage.js) for an example of this.

File: packages/bpk-component-definitions-list/index.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import BpkDefinitionsList from './src/BpkDefinitionsList';
+import BpkDefinition from './src/BpkDefinition';
+import BpkTerm from './src/BpkTerm';
+
+export { BpkDefinitionsList, BpkDefinition, BpkTerm };
```
Comment (matthewdavidson) [inline packages/bpk-component-definitions-list/index.js]: Hey dude - can we rename these to:

- `BpkDefinitionList`
- `BpkDefinitionDescription`
- `BpkDefinitionTerm`

File: packages/bpk-component-definitions-list/package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "bpk-component-definitions-list",
```
Comment (matthewdavidson) [inline packages/bpk-component-definitions-list/package.json]: Can we get rid of the plural here (and all other instances...) ? So just `bpk-component-definition-list`

File: packages/bpk-component-definitions-list/readme.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+## Props
+
+### BpkDefinitionsList
+
+| Property  | PropType | Required | Default Value |
+| --------- | -------- | -------- | ------------- |
+| children  | -        | true     | -             |
+| ordered   | bool     | false    | false         |
```
Comment (matthewdavidson) [inline packages/bpk-component-definitions-list/readme.md]: This `ordered` prop is extraneous right? Copypasta.

File: packages/bpk-component-definitions-list/src/BpkDefinitionsList.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import STYLES from './bpk-definitions-list.scss';
+
+const getClassName = cssModules(STYLES);
+
+const BpkDefinitionsList = (props) => {
+  const classNames = [getClassName('bpk-definitions-list')];
+
+  if (props.className) { classNames.push(props.className); }
+
+  return <dl className={classNames.join(' ')}>{props.children}</dl>;
```
Comment (matthewdavidson) [inline packages/bpk-component-definitions-list/src/BpkDefinitionsList.js]: this should spread arbitrary props like `{...rest}`

File: packages/bpk-component-definitions-list/src/BpkDefinition.js (hunk 1 above)
Comment (matthewdavidson) [reply packages/bpk-component-definitions-list/src/BpkDefinition.js]: this should spread arbitrary props like {...rest}

File: packages/bpk-component-definitions-list/src/BpkTerm.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+import STYLES from './bpk-definitions-list.scss';
+
+const getClassName = cssModules(STYLES);
+
+const BpkTerm = (props) => {
+  const classNames = [getClassName('bpk-definitions-list__term')];
+
+  if (props.className) { classNames.push(props.className); }
+
+  return (<dt className={classNames.join(' ')}>{props.children}</dt>);
```
Comment (matthewdavidson) [inline packages/bpk-component-definitions-list/src/BpkTerm.js]: this should spread arbitrary props like {...rest}

File: packages/bpk-component-definitions-list/src/bpk-definitions-list.scss [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+.bpk-definitions-list {
+  margin-top: 0;
+
+  &__term {
+    font-weight: $bpk-font-weight-bold;
+  }
+
+  &__definition {
+    margin-left: 2rem;
```
Comment (matthewdavidson) [inline packages/bpk-component-definitions-list/src/bpk-definitions-list.scss]: This should be using a token :)

https://backpack.github.io/tokens/spacings/

File: packages/bpk-component-definitions-list/src/BpkDefinition.js (hunk 1 above)
Comment (janeklb) [reply packages/bpk-component-definitions-list/src/BpkDefinition.js]: will do - btw this was lifted from `bpk-component-list` so may want to update there as well

**Comment by janeklb** [comment]
Updated - please have another look. Btw, I'm happy to do the docs for this too, this was just another instance of (no)copypasta as I was basing this off of `bpk-component-list` which has no docs.. afai-can-tell.

**Comment by matteo-hertel** [comment]
Thanks for the PR, looks good!
I have one comment, on a fresh installation I get a different `package-lock` content, that can cause problems during CI, can you please try to run `npm run clean && npm i` and see if you get a different lock as well? if you do please commit that 😄

**Comment by janeklb** [comment]
Thanks @matteo-hertel - indeed I had run `npm i` inside the component's directory, which resulted in that `package-lock.json` file 🤦‍♂️

File: packages/bpk-component-definition-list/src/ComponentFactory.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+
+import STYLES from './bpk-definition-list.scss';
+
+const getClassName = cssModules(STYLES);
+
+const buildComponent = (TagName, baseClassName) => {
```
Comment (k0nserv) [inline packages/bpk-component-definition-list/src/ComponentFactory.js]: Neat approach 👍

**Pull request merged**